pkg/asset/installconfig/aws/validation.go

```diff
@@ -750,8 +761,15 @@ func validateSharedSubnets(ctx context.Context, meta *Metadata, fldPath *field.P
 	}
 
 	for id, subnet := range mergeSubnets(subnets.Private, subnets.Public, subnets.Edge) {
-		if subnet.Tags.HasClusterOwnedTag() {
-			clusterIDs := subnet.Tags.GetOwnedClusterIDs()
+		var clusterIDs []string
+		switch {
+		case subnet.Tags.HasClusterOwnedTag():
```

Comment:
Both of the cases will cause the same function to be run twice. Should we reconstruct this so that it is only run a single time. Something like 
```
if ids := subnet.Tags.GetOwnedClusterIDs(); if len(ids) > 0 {

} else if ...
```

Reply:
Right, I agreed... :+1: 

Though, I purposely made it this way so when reading the code, it's easier to understand where the cluster IDs coming from (i.e. the cluster-owned tags). 

We only ever have max 50 tags on the resource so I am hoping to skip that optimization to allow easier code read :eyes: